server/src/main/java/org/opensearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -129,6 +130,17 @@ public final void collectExistingBucket(LeafBucketCollector subCollector, int do
         subCollector.collect(doc, bucketOrd);
     }
 
+    public final void collectStarTreeBucket(StarTreeBucketCollector collector, long docCount, long bucketOrd, int entryBit)
+        throws IOException {
+        if (docCounts.increment(bucketOrd, docCount) == docCount) {
+            multiBucketConsumer.accept(0);
+        }
```

Comment:
Can you document this method a bit?

In particular, you call `multiBucketConsumer.accept(0)` the first time that you increment the doc count for the given `bucketOrd`. Can you clarify that it's doing that to check the bucket count and memory utilization?